pkg/controller/util/node/controller_utils.go

```diff
@@ -134,9 +135,12 @@ func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
 			continue
 		}
 
-		for i, cond := range pod.Status.Conditions {
+		for _, cond := range pod.Status.Conditions {
 			if cond.Type == v1.PodReady {
-				pod.Status.Conditions[i].Status = v1.ConditionFalse
+				cond.Status = v1.ConditionFalse
+				if !utilpod.UpdatePodCondition(&pod.Status, &cond) {
+					break
```

Comment:
Just confirming that we can be confident that `pod.Status.Conditions` will only contain one `condition` for which `Type = v1.PodReady`? Or alternatively, if there are multiple, we only want to examine one of them?


Confirming my understanding... we can break here because if `UpdatePodCondition` returns false, it means no update is necessary for this condition... and we can be confident that `pod.Status.Condition

Reply:
yeah, only one PodReady condition